### ISSUE TITLE
Shall we remove the non-ssh launchers?

### ISSUE THREAD

**Issue by rhc54** [openpmix/prrte#1308]
This is a copy of an email sent to the PMIx mailing list so those not on that list can comment

-------------------------

After encountering yet another Slurm-induced breakage of the launch system, I find myself wondering about a better long-term solution than continuing to chase the various launch environments. Over the years, we have often encountered problems where the RMs make a change that breaks our integration, often not detected for long periods of time until user complaints reach us. This leads to a complex web of "if-else" clauses as we try to navigate what works for which versions of the RM.

One launch method (ssh/rsh), however, always works. In the dim dark past, there were launch time benefits to using the "native" launcher - but that has not been true for quite a long time now. The ssh tree spawn is generally just as fast as the host environment.

So I'm wondering - should we just remove all these other launchers and always use ssh/rsh?? It would _greatly_ simplify the code and code maintenance, and I can't seen an immediate downside. I realize that some installations have constraints on ssh between various nodes of a cluster, but that (a) is an installation-specific issue, and (b) might be accommodated by adjusting the ssh agent (which the installation is free to specify).

**Comment by rhc54** [comment]
@jsquyres @bwbarrett @mike0042 @awlauria @jjhursey @hppritcha 

You folks are some of the outside stakeholders, so I'm flagging this to your attention. We would retain the support for reading allocations, but would remove the native-based launch mechanisms - thus, we would only support two methods for starting the DVM:

* use ssh for launching the PRRTE daemons. Eliminates many of the problems we have had over the years (whether it be scaling limitations or changing cmd lines or whatever). Essentially matches what MPICH does.
* bootstrap of daemons that startup with the OS. Supports systems using PRRTE as their RTE.

If I don't hear something back by April 9, I'll assume people don't have a problem with this and start removing the PLM components.

**Comment by jjhursey** [comment]
I think you might have missed [my message](https://groups.google.com/g/pmix/c/MQQS8qlssf0/m/ddYWAioaAwAJ) to the mailing list.
 * In some environments the native launcher is the only available mechanism within the allocation. `ssh` is not available/usable within the allocation. This was [echoed](https://groups.google.com/g/pmix/c/MQQS8qlssf0/m/-wQ-HHAbAwAJ) by [others](https://groups.google.com/g/pmix/c/MQQS8qlssf0/m/9ZvPVTkhAwAJ) on the list.
 * We need support for a framework to support a native launch mechanism for the daemons other than `ssh`.
 * Suggestion:
   * Keep the `plm` (and associated) framework.
   * Make the `ssh` component the top priority component.
   * Make the other components supported by the "owner.txt" interested parties. Drop support if they are not maintained.

**Comment by rhc54** [comment]
Wow, that is really disturbing - I received nothing from the mailing list. I'll dig into it. Thx for alerting me!

Based on the mailing list comments, I'll close this for now.

**Comment by hppritcha** [comment]
ssh is not an acceptable solution for some interconnects.  in particular we'd not be able to use SS11 - at least not in a performant manner.

**Comment by hppritcha** [comment]
I did some more checking about the way SS11 handles its authorization keys (OFI) at least for systems running slurm.  It appears the support is handled at a different level than with the older Aries network.  The authorization keys (under the covers not explicitly managed by the application) are available whether or not the process running within the SLURM allocation is a child of slurmd or not.  Verified that with the SSH PLM the CXI provider can be used.
